### ISSUE TITLE
thead > td is not formatted as a header

### ISSUE THREAD

**Issue by andrewnicols** [twbs/bootstrap#37586]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

The `thead` tag should cause all child content to be treated as a header, however if you write a table with a `<td>` within the `<thead>` tag, it is treated differently to a `<th>` tag.

This topic is discussed within the W3C HTML specification in [this issue](https://github.com/w3c/html/issues/819) where it is declared that any content of a thead should be treated as a header. The wording of the specification is amended to:

> The thead element represents the block of rows that consist of the column labels (headers) for the parent table element, if the thead element has a parent and it is a table.

https://html.spec.whatwg.org/#the-thead-element

The apparent intent (from the bug above) is that all content of a `<thead>` tag should act as, and appear as, a header.

At present this is not the behaviour and a `thead > td` appears differently to a `thead > th`.

### Reduced test cases

https://codepen.io/andrewnicols/pen/vYrvXBX

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome, Safari, Firefox, Microsoft Edge

### What version of Bootstrap are you using?

4.6.0; 5.2.x

**Comment by patrickhlauke** [comment]
as the original discussion/bug on the standard didn't really get resolved, I started a fresh bug here https://github.com/whatwg/html/issues/8650

splitting out the aspect of whether it structurally works correctly as a header cell or not from the visual presentation, though, I could see some merit in visually, at least, treating `thead > td` the same as `thead > tr`

**Comment by Sarvesh-Ghildiyal** [comment]
@andrewnicols  <thead> tag is used inside <table> to specify **header part of the table** similar tags are <tbody> for **table body** and <tfoot> for **table footer**. Now, <th> and <td> are usual table tags where <th> is used for **table heading** and <td> is used for **table data**. So, obviously when u write a something inside <th> _its gonna get bold and highlighted_ where as text inside <td> will be of usual self. Also, it is not the matter if u have used <th> and <td> inside <thead> or not they gonna act their usual self, _table heading and table data_. Hope! this helps u..

**Comment by andrewnicols** [comment]
Hi @Sarvesh-Ghildiyal ,

Thanks for the response.

Whilst you can indeed argue that a `<td>` inside of a `<thead>` should be considered as table data because that's what ` <td>` element does, that isn't what the W3C spec states, again:

> The thead element represents the block of rows that consist of the column labels (headers) for the parent table element, if the thead element has a parent and it is a table.

This does not indicate anywhere that a `thead` should only include `th` for them to be considered as header content. Any content of a `thead` should be considered to be header content.

**Comment by patrickhlauke** [comment]
Coming back to this, from the recent renewed comments on https://github.com/whatwg/html/issues/8650 ... it seems clear that the HTML spec itself does not intend for `<td>` elements inside `<thead>` to *become* table headers, semantically. the very example in the spec at https://html.spec.whatwg.org/#the-thead-element shows that. on the subject of semantics, i've now filed a small tweak/patch to the HTML spec itself to dismabiguate this.

As for the issue here, I would say the styling of a `<td>` inside `<thead>` should *not* match the style of a `<th>`. This is not the case for the browser's own vanilla default stylesheets, and should not be the case for bootstrap's styles. If an author wishes to make their `<td>`s look like `<th>`s, they can achieve this with additional utility classes.
